src/main/java/com/mineboundteam/minebound/magic/OffensiveSpells/NecroticOffensiveSpell.java

```diff
@@ -0,0 +1,170 @@
+    private final double damageBoost;
+    private final double vampirePercent;
+
+    public final static String ACTIVE_TAG = MineBound.MOD_ID + ".necrotic_offensive.active";
+    public final static String TICK_TAG = MineBound.MOD_ID + ".necrotic_offensive.ticks";
+
+    public NecroticOffensiveSpell(Properties properties, NecroticOffensiveSpellConfig config) {
+        super(properties, config.LEVEL, MagicType.NECROTIC, SpellType.OFFENSIVE);
+        manaCost = config.MANA_COST.get();
+        damageBoost = config.DMG_BOOST.get();
+        vampirePercent = config.HEALTH_AMT.get();
+    }
+
+    @Override
+    public void use(ItemStack stack, Level level, Player player) {
+        stack.getOrCreateTag().putBoolean(ACTIVE_TAG, true);
+    }
+
+    @Override
+    public void onUsingTick(ItemStack stack, Level level, Player player) {
+        if(stack.getOrCreateTag().getBoolean(ACTIVE_TAG)) {
+            int ticks = stack.getOrCreateTag().getInt(TICK_TAG);
+            if(ticks < 20)
+                ticks++;
+            else {
+                reduceMana(manaCost, player);
+                ticks = 0; 
+            }
+            stack.getOrCreateTag().putInt(TICK_TAG, ticks);
+        }
+    }
+
+    @Override
+    public void releaseUsing(ItemStack stack, Level level, Player player) {
+        stack.getOrCreateTag().putBoolean(ACTIVE_TAG, false);
+        stack.getOrCreateTag().putInt(TICK_TAG, 0);
+    }
+
+    @SubscribeEvent
+    public static void onAttack(LivingDamageEvent event){
+        if(event.getSource().getDirectEntity() != null && event.getSource().getDirectEntity() instanceof ServerPlayer player){
+            ItemStack spell = ItemStack.EMPTY; 
+            if(player.getUseItem().getItem() instanceof NecroticOffensiveSpell)
+                spell = player.getUseItem();
+            else {
+                if(player.getCapability(PlayerSelectedSpellsProvider.PRIMARY_SPELL).isPresent()){
+                    PrimarySelected selected = player.getCapability(PlayerSelectedSpellsProvider.PRIMARY_SPELL).resolve().get(); 
+                    if(!selected.isEmpty())
+                        spell = ItemStack.of(ArmorNBTHelper.getSpellTag(player.getItemBySlot(selected.equippedSlot), ArmorNBTHelper.ACTIVE_SPELL).getCompound(selected.index));
+                }
+                if(player.getCapability(PlayerSelectedSpellsProvider.SECONDARY_SPELL).isPresent()){
+                    SecondarySelected selected = player.getCapability(PlayerSelectedSpellsProvider.SECONDARY_SPELL).resolve().get(); 
+                    if(!selected.isEmpty())
+                        spell = ItemStack.of(ArmorNBTHelper.getSpellTag(player.getItemBySlot(selected.equippedSlot), ArmorNBTHelper.ACTIVE_SPELL).getCompound(selected.index));
+                }
+            }
+
+        
+            if(spell.getOrCreateTag().getBoolean(ACTIVE_TAG) && spell.getItem() instanceof NecroticOffensiveSpell spellItem) {
+                double amount = event.getAmount();
```

Comment:
Is there a reason you are storing `event.getAmount()` as a double and then casting back to a float when calling `setAmount`?

Reply:
Fixed